Doc/README.rst

```diff
@@ -7,45 +7,49 @@ available <https://docs.python.org/dev/download.html>`_.
 
 Documentation on authoring Python documentation, including information about
 both style and markup, is available in the "`Documenting Python
-<https://docs.python.org/devguide/documenting.html>`_" chapter of the
+<https://devguide.python.org/documenting/>`_" chapter of the
 developers guide.
 
 
 Building the docs
 =================
 
-You need to have `Sphinx <http://sphinx-doc.org/>`_ installed; it is the toolset
-used to build the docs.  It is not included in this tree, but maintained
-separately and `available from PyPI <https://pypi.python.org/pypi/Sphinx>`_.
+The documentation is built with several tools which are not included in this
+tree but are maintained separately and are available from
+`PyPI <https://pypi.org/>`_.
 
+* `Sphinx <https://pypi.org/project/Sphinx/>`_
+* `blurb <https://pypi.org/project/blurb/>`_
 
-Using make
-----------
+The easiest way to install these tools is to create a virtual environment and
+install the tools into there.
 
-A Makefile has been prepared so that on Unix, provided you have installed
-Sphinx, you can just run ::
 
-   make html
-
-to build the HTML output files.
+Using make
+----------
 
-On Windows, we try to emulate the Makefile as closely as possible with a
-``make.bat`` file.
+To get started on UNIX, you can create a virtual environment with the command ::
 
-To use a Python interpreter that's not called ``python``, use the standard
-way to set Makefile variables, using e.g. ::
+  make venv
 
-   make html PYTHON=python3
+That will install all the tools necessary to build the documentation. Assuming
+the virtual environment was created in the ``env`` directory (the default;
+configurable with the VENVDIR variable), you can run the following command to
+build the HTML output files::
 
-On Windows, set the PYTHON environment variable instead.
+  make html PYTHON=./env/bin/python
```

Comment:
I know I suggested something similar for blurb.  But now I'm not sure that was the best choice.  It's more problematic with sphinx-build since existing doc builds may have been depending on using an already installed (e.g. system) sphinx-build and have never bothered before to run the `make venv` step (and before, even if they did run that step, it wouldn't have made a difference without tweaking `PYTHON`).  With this change, those builds would now fail unless sphinx had happened to be installed in whatever `python3` on PATH points to.  I don't know how many users this would affect but it's not hypothetical: I ran into this myself.  There are several ways around it; I was thinking the most straightforward way might be to play with PATH in the "build" rule; so something like:
  `PATH=./venv/bin:$PATH  sphinx-build`

Either that or we need to make sure this is documented as an incompatible change and then maybe only for 3.7?